### PR TITLE
added NextCount

### PR DESCRIPTION
**Pull request by WIZARDISHUNGRY**
I added functionality for allowing the user to specify a custom nonce counter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,3 +118,46 @@ func main() {
   req2.Header.Set("Authorization", cred.String())
 }
 ```
+
+## Custom Nonce Counter
+
+If the service does not provide unique nonces for multiple clients and you must not duplicate nonce counts.
```
Comment (icholy) [inline README.md:124]: Seems like it would be easier to just get a new challenge for every request. Something like a `NoCache` flag.

Comment (WIZARDISHUNGRY) [reply README.md:124]: [Akamai Media Services for Live](https://learn.akamai.com/en-us/webhelp/media-services-live/media-services-live-encoder-compatibility-testing-and-qualification-guide-v4.0/GUID-8CD20D40-9629-41A4-B7F8-D6EA408BC417.html) sends all clients the same challenge (identical Opaque + Nonce values), rotating each hour. 🙃

Comment (icholy) [reply README.md:124]: dang, ok.

**Comment by icholy** [comment]
What about something more general purpose?

``` go
tr := &digest.Transport{
  Username: "foo",
  Password: "bar",
  Digest: func(req *http.Request, chal *digest.Challenge) (*digest.Credentials, error) {
    chal.Count = 123
    return digest.Digest(chal)
  },
}

```

**Comment by icholy** [comment]
I've pushed `v0.1.13` with  `Transport.Digest`. It should cover your use-case.

**Comment by icholy** [comment]
Sorry I didn't merge as-is. You're code is solid, but the api was too narrow.

**Comment by WIZARDISHUNGRY** [comment]
Yep, this is very good. Thank you!

**Pull request closed without merging**